src/swaps/oneinch/OneinchSwapProvider.js

```diff
@@ -0,0 +1,263 @@
+    })
+    await this.sendLedgerNotification(quote, account, 'Signing required to complete the swap.')
+    const swapTx = await client.chain.sendTransaction(trade.data.tx)
+    return {
+      status: 'WAITING_FOR_SWAP_CONFIRMATIONS',
+      swapTx,
+      swapTxHash: swapTx.hash
+    }
+  }
+
+  async newSwap ({ network, walletId, quote }) {
+    const approvalRequired = isERC20(quote.from)
+    const updates = approvalRequired
+      ? await this.approveTokens({ network, walletId, quote })
+      : await this.sendSwap({ network, walletId, quote })
+
+    return {
+      id: uuidv4(),
+      fee: quote.fee,
+      slippage: 50,
+      ...updates
+    }
+  }
+
+  async estimateFees ({ network, walletId, asset, accountId, txType, quote, feePrices, max }) {
+    if (txType in OneinchSwapProvider.feeUnits) {
+      const fees = {}
+      for (const feePrice of feePrices) {
+        fees[feePrice] = getTxFee(OneinchSwapProvider.feeUnits[txType], asset, feePrice)
+      }
+      return fees
+    }
+  }
+
+  async waitForApproveConfirmations ({ swap, network, walletId }) {
+    const account = this.getAccount(swap.accountId)
+    const client = this.getClient(network, walletId, swap.from, account?.type)
+
+    try {
+      const tx = await client.chain.getTransactionByHash(swap.approveTxHash)
+      if (tx && tx.confirmations > 0) {
+        return {
+          endTime: Date.now(),
+          status: 'APPROVE_CONFIRMED'
+        }
+      }
+    } catch (e) {
+      if (e.name === 'TxNotFoundError') console.warn(e)
+      else throw e
+    }
+  }
+
+  async waitForSwapConfirmations ({ swap, network, walletId }) {
+    const account = this.getAccount(swap.accountId)
+    const client = this.getClient(network, walletId, swap.from, account?.type)
+
+    try {
+      const tx = await client.chain.getTransactionByHash(swap.swapTxHash)
+      if (tx && tx.confirmations > 0) {
+        this.updateBalances({ network, walletId, assets: [swap.from] })
+        return {
+          endTime: Date.now(),
+          status: 'SUCCESS'
+        }
+      }
+    } catch (e) {
+      if (e.name === 'TxNotFoundError') console.warn(e)
+      else throw e
+    }
+  }
+
+  async performNextSwapAction (store, { network, walletId, swap }) {
+    let updates
+
+    switch (swap.status) {
+      case 'WAITING_FOR_APPROVE_CONFIRMATIONS':
+        updates = await withInterval(async () => this.waitForApproveConfirmations({ swap, network, walletId }))
+        break
+      case 'APPROVE_CONFIRMED':
+        updates = await withLock(store, { item: swap, network, walletId, asset: swap.from },
+          async () => this.sendSwap({ quote: swap, network, walletId }))
+        break
+      case 'WAITING_FOR_SWAP_CONFIRMATIONS':
+        updates = await withInterval(async () => this.waitForSwapConfirmations({ swap, network, walletId }))
+        break
+    }
+
+    return updates
+  }
+
+  static txTypes = {
+    SWAP: 'SWAP'
+  }
+
+  static feeUnits = {
```

Comment:
this should be removed after fee estimate comes from quote

Reply:
will remove it